src/swupd/manifest_test.go

```diff
@@ -180,3 +180,81 @@ func TestReadManifestFileEntry(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckHeaderPopulated(t *testing.T) {
+	t.Run("populated", func(t *testing.T) {
+		m := Manifest{
+			Header: ManifestHeader{
+				Format:      10,
+				Version:     100,
+				Previous:    90,
+				FileCount:   553,
+				ContentSize: 100000,
+				TimeStamp:   time.Unix(1000, 0),
+				// does not fail when includes not added
+			},
+		}
+
+		if err := m.CheckHeaderPopulated(); err != nil {
+			t.Error("CheckHeaderPopulated raised error for valid header")
+		}
+	})
+
+	zeroTime := time.Time{}
+	var includes []Manifest
+	invalidHeaders := []ManifestHeader{
+		{0, 100, 90, 553, time.Unix(1000, 0), 100000, includes},
+		{10, 0, 90, 553, time.Unix(1000, 0), 100000, includes},
+		{10, 100, 0, 553, time.Unix(1000, 0), 100000, includes},
+		{10, 100, 90, 0, time.Unix(1000, 0), 100000, includes},
+		{10, 100, 90, 553, zeroTime, 100000, includes},
+		{10, 100, 90, 553, time.Unix(1000, 0), 0, includes},
+	}
+
+	for _, header := range invalidHeaders {
+		t.Run("unpopulated", func(t *testing.T) {
+			m := Manifest{Header: header}
+			if err := m.CheckHeaderPopulated(); err == nil {
+				t.Error("CheckHeaderPopulated did not return an error on invalid header")
+			}
+		})
+	}
+}
+
+func TestReadManifestFromFile(t *testing.T) {
+	t.Run("good manifest", func(t *testing.T) {
+		path := "testfiles/manifest.good"
```

Comment:
The convention is to use testdata/ directory instead of testfiles/. Go tool will not walk through that directory when looking for sources (not much practical difference for us, but still).